File: empty_folders.pl
use strict;
use warnings;
use File::Find;

my $path = '/';
find( \&delete_empty_folders, $path);

sub delete_empty_folders {
    if (-d) {
        if (! -e "$_/*") {
            rmdir $_;
            print "Deleted: ", $_, "

";
        }
    }
}

File: empty_folders.py
import os 

def delete_empty_folders():
    for root, dirs, files in os.walk("/"):
        for dir in dirs:
            if not os.listdir(os.path.join(root, dir)):
                os.rmdir(os.path.join(root, dir))
                print("Deleted: ", os.path.join(root, dir))
delete_empty_folders()